contracts/rateproducer/rateproducer.cpp

```diff
@@ -4,257 +4,470 @@
+               infrastructure_cntr++;
+           }
+            
+           if(bps_it->trustiness){
+               trustiness_total+=bps_it->trustiness;
+               trustiness_cntr++;
+           }
+            
+           if(bps_it->community){
+               community_total+=bps_it->community;
+               community_cntr++;
+           }
+          
+           if(bps_it->development){
+               development_total+=bps_it->development;
+               development_cntr++;
+           }
+           bps_it ++;
+        }
+        
+        if(transparency_cntr){
+            *transparency = transparency_total/transparency_cntr;
+            category_counter++;
+        }
+        
+        if(infrastructure_cntr){
+            *infrastructure =infrastructure_total/infrastructure_cntr;
+            category_counter++;
+        }
+           
+        if(trustiness_cntr){
+            *trustiness = trustiness_total/trustiness_cntr;
+            category_counter++;
+        }
+           
+        if(community_cntr){
+            *community = community_total/community_cntr;
+            category_counter++;
+        }
+           
+        if(development_cntr){
+            *development = development_total/development_cntr;
+            category_counter++;
+        } 
+        *average = (*transparency + *infrastructure + *trustiness + *community +*development)/category_counter;
+        *ratings_cntr = std::max(transparency_cntr,std::max(infrastructure_cntr,std::max(trustiness_cntr,std::max(community_cntr,development_cntr))));
+    }
 
-      if(!table.compare("stats")){
-      	producers_stats_table bps_stats(_self, _self.value);
-      	auto itr_stats = bps_stats.begin();
-      	while ( itr_stats != bps_stats.end()) {
+    
+    ACTION erase(name bp_name) {
+        
+        require_auth(_self);
+
+        producers_table bps(_self, _self.value);
+        auto itr = bps.begin();
+        while ( itr != bps.end()) {
+            if(itr->bp == bp_name){
+                itr = bps.erase(itr);
+            }else{
+                itr++;
+            }
+        }
+
+        producers_stats_table bps_stats(_self, _self.value);
+        auto itr_stats = bps_stats.begin();
+        while ( itr_stats != bps_stats.end()) {
+            if(itr_stats->bp == bp_name){
+                itr_stats = bps_stats.erase(itr_stats);
+            }else{
+                itr_stats++;
+            }
+        }
+    }
+    
+    ACTION wipe() {
+        
+        require_auth(_self);
+        producers_table bps(_self, _self.value);
+        auto itr = bps.begin();
+        while ( itr != bps.end()) {
+            itr = bps.erase(itr);
+        }
+
+        producers_stats_table bps_stats(_self, _self.value);
+        auto itr_stats = bps_stats.begin();
+        while ( itr_stats != bps_stats.end()) {
             itr_stats = bps_stats.erase(itr_stats);
-      	}
-      }
+        }
     }
 
+
   private:
     TABLE block_producers_stats {
```

Comment:
@kriskoin  I am wondering if this table would be more apropriately named simply `stats`

Reply:
@xavier506  sure, i'll changed

Reply:
@xavier506  done!!